Lib/test/test_inspect/test_inspect.py

```diff
@@ -1816,49 +1816,109 @@ def test_typing_replacement(self):
 class TestIsMethodDescriptor(unittest.TestCase):
 
     def test_custom_descriptors(self):
+        # Custom method descriptors (non-data descriptors):
+
         class MethodDescriptor:
+            """with __get__ and no __set__/__delete__"""
             def __get__(self, *_): pass
+        class MethodDescriptorWithGetNone:
+            """with __get__=None and no __set__/__delete__"""
+            __get__ = None
         class MethodDescriptorSub(MethodDescriptor):
-            pass
-        class DataDescriptorWithNoGet:
+            """with __get__ (inherited) and no __set__/__delete__"""
+
+        for cls in [
+            MethodDescriptor,
+            MethodDescriptorWithGetNone,
+            MethodDescriptorSub,
+        ]:
+            self.assertTrue(
+                inspect.ismethoddescriptor(cls()),
+                f"an object {cls.__doc__} is a method descriptor",
+            )
+            self.assertFalse(
+                inspect.ismethoddescriptor(cls),
+                f'a class of a method descriptor is not a method descriptor',
+            )
```

Comment:
Use `with self.subTest(cls=cls):` instead of an error message in the assert.

Reply:
OK, I'll do that.

I wasn't sure whether it is an approved practice in the stdlib tests, but now I am encouraged. :)

Reply:
Old code wasn't using this practice, but new code is generally using it (at least from what I could saw in the other PRs).